tests/smoke_test.go

```diff
@@ -52,6 +53,38 @@ func TestLanguageNewSmoke(t *testing.T) {
 	}
 }
 
+// Quick sanity tests that YAML convert works.
+func TestYamlConvertSmoke(t *testing.T) {
+	t.Parallel()
+
+	e := ptesting.NewEnvironment(t)
+	defer deleteIfNotFailed(e)
+
+	e.ImportDirectory("testdata/random_yaml")
+
+	// Make sure random is installed
+	out, _ := e.RunCommand("pulumi", "plugin", "ls")
+	if !strings.Contains(out, "random  resource  4.13.0") {
+		e.RunCommand("pulumi", "plugin", "install", "resource", "random", "4.13.0")
+	}
+
+	e.RunCommand(
+		"pulumi", "convert", "--strict",
+		"--language", "pcl", "--from", "yaml", "--out", "out")
+
+	actualPcl, err := os.ReadFile(filepath.Join(e.RootPath, "out", "program.pp"))
```

Comment:
if `pulumi convert` from yaml decides later that the generated file is `main.pp` instead of `program.pp` this will fail. We could get the files from the `out` directory and read the first one with a `.pp` extension. Not a blocker for this PR though

Reply:
I think that's fine, I'm ok with this being a little tightly coupled it will only change when we bump the version of YAML that we reference.